### PR TITLE
[TACACS][IPv6]: Enable IPv6 in ptf-docker and add TACACS IPv6 test case

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Enable IPv6 in ptf-docker, to test TACACS on IPv6.

#### How did you do it?
- Add ptf_ipv6 parameter which will have ipv6 address in veos and vtestbed.csv
- Add ansible_hostv6 parameter which will have IPv6 address for DUT; this is done in “lab” inventory file.
- Changes in parser to accept above changes.
- Change to add mgmt. v6 address to ptf-docker
- Change in minigraph template to use the ansible_hostv6 address provided in the inventory file.
- Modify tacacs test case to run with ipv6 address.

#### How did you verify/test it?
- Bring up VS testbed
- ptf-docker should come up with IPV6 mgmt IP address and sonic-vs image should come up with ipv6 address.
- Run tacacs test.

### CODE REVIEW

File: tests/vtestbed.csv [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# conf-name,group-name,topo,ptf_image_name,ptf,ptf_ip,server,vm_base,dut,comment
-vms-kvm-t0,vms6-1,t0,docker-ptf,ptf-01,10.250.0.102/24,server_1,VM0100,[vlab-01],Tests virtual switch vm
-vms-kvm-t0-64,vms6-1,t0-64,docker-ptf,ptf-01,10.250.0.102/24,server_1,VM0100,[vlab-02],Tests virtual switch vm
-vms-kvm-t1-lag,vms6-2,t1-lag,docker-ptf,ptf-02,10.250.0.106/24,server_1,VM0104,[vlab-03],Tests virtual switch vm
-vms-kvm-t0-2,vms6-3,t0,docker-ptf,ptf-03,10.250.0.108/24,server_1,VM0104,[vlab-04],Tests virtual switch vm
+# conf-name,group-name,topo,ptf_image_name,ptf,ptf_ip,ptf_ipv6,server,vm_base,dut,comment
+vms-kvm-t0,vms6-1,t0,docker-ptf,ptf-01,10.250.0.102/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0100,[vlab-01],Tests virtual switch vm
+vms-kvm-t0-64,vms6-1,t0-64,docker-ptf,ptf-01,10.250.0.102/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0100,[vlab-02],Tests virtual switch vm
+vms-kvm-t1-lag,vms6-2,t1-lag,docker-ptf,ptf-02,10.250.0.106/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0104,[vlab-03],Tests virtual switch vm
+vms-kvm-t0-2,vms6-3,t0,docker-ptf,ptf-03,10.250.0.108/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0104,[vlab-04],Tests virtual switch vm
```
Comment (neethajohn) [inline tests/vtestbed.csv:5]: same ipv6 address?

File: ansible/testbed_add_vm_topology.yml [hunk 2]
```diff
@@ -53,6 +54,10 @@
     fail: msg="Define ptf ip variable with -e ptf_ip=something"
     when: ptf_ip is not defined
 
+  - name: Check that variable ptf_ipv6 is defined
+    fail: msg="Define ptf ip variable with -e ptf_ipv6=something"
```
[the remark this answers is not in the document]
Comment (SuvarnaMeenakshi) [reply ansible/testbed_add_vm_topology.yml]: Fixed

File: ansible/vtestbed.csv [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# conf-name,group-name,topo,ptf_image_name,ptf,ptf_ip,server,vm_base,dut,comment
-vms-kvm-t0,vms6-1,t0,docker-ptf,ptf-unknown,10.250.0.102/24,server_1,VM0100,[vlab-01],Tests virtual switch vm
-vms-kvm-t0-64,vms6-1,t0-64,docker-ptf,ptf-unknown,10.250.0.102/24,server_1,VM0100,[vlab-02],Tests virtual switch vm
-vms-kvm-t1-lag,vms6-2,t1-lag,docker-ptf,ptf-unknown,10.250.0.106/24,server_1,VM0104,[vlab-03],Tests virtual switch vm
-vms-kvm-t0-2,vms6-3,t0,docker-ptf,ptf-unknown,10.250.0.108/24,server_1,VM0104,[vlab-04],Tests virtual switch vm
+# conf-name,group-name,topo,ptf_image_name,ptf,ptf_ip,ptf_ipv6,server,vm_base,dut,comment
+vms-kvm-t0,vms6-1,t0,docker-ptf,ptf-unknown,10.250.0.102/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0100,[vlab-01],Tests virtual switch vm
+vms-kvm-t0-64,vms6-1,t0-64,docker-ptf,ptf-unknown,10.250.0.102/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0100,[vlab-02],Tests virtual switch vm
+vms-kvm-t1-lag,vms6-2,t1-lag,docker-ptf,ptf-unknown,10.250.0.106/24,fec0:0:0:0:0:ffff:afa:2/64,server_1,VM0104,[vlab-03],Tests virtual switch vm
```
[the remark this answers is not in the document]
Comment (SuvarnaMeenakshi) [reply ansible/vtestbed.csv]: Fixed.

File: ansible/vtestbed.csv [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
+vms-kvm-t0-2,vms6-3,t0,docker-ptf,ptf-unknown,10.250.0.108/24,,server_1,VM0104,[vlab-04],Tests virtual switch vm
```
[the remark this answers is not in the document]
Comment (SuvarnaMeenakshi) [reply ansible/vtestbed.csv]: Added ipv6 address.

File: tests/vtestbed.csv (hunk 1 above)
Comment (SuvarnaMeenakshi) [reply tests/vtestbed.csv]: Corrected, provided ip addresses same as in ansible/vtestbed.csv

**Comment by yxieca** [comment]
retest vsimage please

**Comment by lguohan** [comment]
I am curious why this test does not run in the kvmtest?  https://sonic-jenkins.westus2.cloudapp.azure.com/job/common/job/sonic-mgmt-testing/job/sonic-mgmt-pr/392/testReport/tacacs/test_ro_user/test_ro_user/

no ipv6 version run.

**Comment by lguohan** [comment]
since you add dut ipv6 address, I think you need to add this ipv6 to the minigraph.

https://github.com/Azure/sonic-mgmt/blob/master/ansible/templates/minigraph_dpg.j2#L35

File: ansible/lab [hunk 5]
```diff
@@ -6,10 +6,10 @@ hwsku="ACS-MSN2700"
 iface_speed='40000'
 
 [sonic_s6000]
-lab-s6000-01      ansible_host=10.251.0.189
-vlab-01           ansible_host=10.250.0.101
-vlab-03           ansible_host=10.250.0.105
-vlab-04           ansible_host=10.250.0.107
+lab-s6000-01      ansible_host=10.251.0.189    ansible_hostv6=fec0:0:0:0:0:ffff:afa:9
```
Comment (lguohan) [inline ansible/lab:9]: this can be simplified as fec0::ffff:afa:9

File: tests/veos.vtb [hunk 6]
```diff
@@ -61,11 +61,11 @@ vlab-02
 vlab-03
 
 [sonic]
-vlab-01 ansible_host=10.250.0.101 type=kvm hwsku=Force10-S6000 ansible_password=password ansible_user=admin
-vlab-02 ansible_host=10.250.0.102 type=kvm hwsku=Force10-S6100 ansible_password=password
-vlab-03 ansible_host=10.250.0.105 type=kvm hwsku=Force10-S6000 ansible_password=password ansible_user=admin
-vlab-04 ansible_host=10.250.0.107 type=kvm hwsku=Force10-S6000 ansible_password=password ansible_user=admin
+vlab-01 ansible_host=10.250.0.101 ansible_hostv6=fec0:0:0:0:0:ffff:afa:1 type=kvm hwsku=Force10-S6000 ansible_password=password ansible_user=admin
```
Comment (lguohan) [inline tests/veos.vtb:64]: simplify the v6 ip address: fec0::ffff:afa:1

File: ansible/testbed_refresh_dut.yml [hunk 7]
```diff
@@ -24,6 +24,7 @@
 # -e dut_name=str-msn2700-01 - the name of target dut
 # -e VM_base=VM0300          - the VM name which is used to as base to calculate VM name for this set
 # -e ptf_ip=10.255.0.255/23 - the ip address and prefix of ptf container mgmt interface
+# -e ptf_ipv6=fec0:0:0:0:0:ffff:afa:1/64 - the ipv6 address and prefix of ptf container mgmt interface
```
Comment (lguohan) [inline ansible/testbed_refresh_dut.yml:27]: simplify the ipv6 address

**Comment by SuvarnaMeenakshi** [comment]
> I am curious why this test does not run in the kvmtest? https://sonic-jenkins.westus2.cloudapp.azure.com/job/common/job/sonic-mgmt-testing/job/sonic-mgmt-pr/392/testReport/tacacs/test_ro_user/test_ro_user/
> 
> no ipv6 version run.
> 
> @daall , do you know why?

test_ro_user does not have new function added to test with ipv6, Will update the test file.

**Comment by yxieca** [comment]
retest vsimage please

File: ansible/lab (hunk 5 above)
Comment (SuvarnaMeenakshi) [reply ansible/lab]: fixed as per comment.

File: tests/veos.vtb (hunk 6 above)
Comment (SuvarnaMeenakshi) [reply tests/veos.vtb]: fixed as per comment.

File: ansible/testbed_refresh_dut.yml (hunk 7 above)
Comment (SuvarnaMeenakshi) [reply ansible/testbed_refresh_dut.yml]: fixed as per comment.

File: tests/common/plugins/tacacs.py [hunk 8]
```diff
@@ -1,8 +1,7 @@
 import pytest
 import crypt
 
-@pytest.fixture(scope="module")
-def setup_tacacs(ptfhost, duthost, creds):
+def start_tacacs(ptfhost, duthost, creds, tacacs_server_ip):
```
Comment (lguohan) [inline tests/common/plugins/tacacs.py:4]: i prefer the previous name, setup_tacacs. it means setup the tacacs on both client and server.

Comment (SuvarnaMeenakshi) [reply tests/common/plugins/tacacs.py]: Modified as per comment.

**Comment by lguohan** [comment]
retest this please

**Comment by SuvarnaMeenakshi** [comment]
retest this please

**Comment by NazarTkachuk** [comment]
@here `TestbedProcessing.py` is broken.
see traceback bellow
```
PROCESS STARTED
BACKUP PROCESS STARTED
Error: could not back up group_vars/eos/creds.yml
BACKUP PROCESS COMPLETED
LOADING PROCESS STARTED
LOADING: bfn-testbed.yaml
TestbedProcessing.py:532: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  doc = yaml.load(open(args.i, 'r'))
LOADING PROCESS COMPLETED
GENERATING FILES FROM CONFIG FILE
        CREATING SONIC LAB LINKS: files/sonic_lab_links.csv
        CREATING SONIC LAB DEVICES: files/sonic_lab_devices.csv
        CREATING TEST BED: testbed.csv
Traceback (most recent call last):
  File "TestbedProcessing.py", line 583, in <module>
    main()
  File "TestbedProcessing.py", line 559, in main
    makeTestbed(testbed, args.basedir + testbed_file)  # Generate testbed.csv (TESTBED)
  File "TestbedProcessing.py", line 230, in makeTestbed
    if not ptf_ipv6:
UnboundLocalError: local variable 'ptf_ipv6' referenced before assignment
```

**Pull request merged**